src/modules/ModuleTradeShared/composable.pair-input.ts

```diff
@@ -24,6 +27,56 @@ export function useLocalStorageAddrsOrigin(key: string, isActive?: Ref<boolean>)
   return offable
 }
 
+export function useAddrRouteParams({
+  router,
+  baseToken,
+  additionalBaseToken,
+  isActive,
+}: {
+  router: Router
+  baseToken?: Address
+  additionalBaseToken?: Address
+  isActive?: Ref<boolean>
+}): { pair: Readonly<Ref<TokensPair<Address | null>>>; clear: () => void } {
+  const params = {
+    tokenA: useRouteParams('tokenA'),
+    tokenB: useRouteParams('tokenB'),
+  }
+
+  const pair = computed<TokensPair<Address | null>>(() => {
```

Comment:
I still don't understand why you prefer `TokensPair<Address | null>` over `null | TokensPair<Address>`.

Also, you could use conditional types to return `{ tokenA: Address, tokenB: null | Address }` in case when `additionalBaseToken` is not provided.

Reply:
Replaced with ```null | { tokenA: Address, tokenB: null | Address }```